python/lsst/ip/diffim/getTemplate.py

```diff
@@ -76,17 +88,97 @@ def run(self, exposure, sensorRef, templateIdList=None):
             - ``sources`` :  None for this subtask
         """
         skyMap = sensorRef.get(datasetType=self.config.coaddName + "Coadd_skyMap")
+        tractInfo, patchList, skyCorners = self.getOverlapPatchList(exposure, skyMap)
+
+        availableCoadds = dict()
+        for patchInfo in patchList:
+            patchNumber = tractInfo.getSequentialPatchIndex(patchInfo)
+            patchArgDict = dict(
+                datasetType=self.getCoaddDatasetName() + "_sub",
+                bbox=patchInfo.getOuterBBox(),
+                tract=tractInfo.getId(),
+                patch="%s,%s" % (patchInfo.getIndex()[0], patchInfo.getIndex()[1]),
+                numSubfilters=self.config.numSubfilters,
+            )
+
+            if self.config.coaddName != 'dcr' and sensorRef.datasetExists(**patchArgDict):
+                self.log.info("Reading patch %s" % patchArgDict)
+                availableCoadds[patchNumber] = sensorRef.get(**patchArgDict)
+
+        templateExposure = self.makeTemplateExposure(
+            tractInfo, patchList, skyCorners, availableCoadds,
+            sensorRef=sensorRef, visitInfo=exposure.getInfo().getVisitInfo()
+        )
+        return pipeBase.Struct(exposure=templateExposure, sources=None)
+
+    def runGen3(self, exposure, butlerQC, skyMapRef, coaddExposureRefs):
+        """Gen3 task entry point. Retrieve and mosaic a template coadd exposure
+        that overlaps the science exposure.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The science exposure to define the sky region of the template coadd.
+        butlerQC : `lsst.pipe.base.ButlerQuantumContext`
+            Butler like object that supports getting data by DatasetRef.
+        skyMapRef : `lsst.daf.butler.DatasetRef`
+            Reference to SkyMap object that corresponds to the template coadd.
+        coaddExposureRefs : iterable of `lsst.daf.butler.DatasetRef`
+            Iterable of references to the available template coadd patches.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            - ``exposure`` : `lsst.afw.image.ExposureF`
+                a template coadd exposure assembled out of patches
+            - ``sources`` :  `None` for this subtask
+        """
+        skyMap = butlerQC.get(skyMapRef)
+        tractInfo, patchList, skyCorners = self.getOverlapPatchList(exposure, skyMap)
+        patchNumFilter = frozenset(tractInfo.getSequentialPatchIndex(p) for p in patchList)
+
+        availableCoadds = dict()
+        for coaddRef in coaddExposureRefs:
+            dataId = coaddRef.dataId
+            if dataId['tract'] == tractInfo.getId() and dataId['patch'] in patchNumFilter:
+                self.log.info("Using template input tract=%s, patch=%s" %
+                              (tractInfo.getId(), dataId['patch']))
+                availableCoadds[dataId['patch']] = butlerQC.get(coaddRef)
```

Comment:
We don't want to load all of the coadds from disk here, so I think you just want something like:
```
    availableCoaddRefs = dict()
    for coaddRef in coaddExposureRefs:
         dataId = coaddRef.dataId
          if dataId['tract'] == tractInfo.getId() and dataId['patch'] in patchNumFilter:
             self.log.info("Using template input tract=%s, patch=%s" %
                                 (tractInfo.getId(), dataId['patch']))
             availableCoaddRefs[dataId['patch']] = coaddRef
```
Then pass `availableCoaddRefs ` to `run` instead of `availableCoadds`